core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3026,6 +3015,183 @@ private void optimiseOrderBy(QueryModel model, int topLevelOrderByMnemonic) {
+        // loop over the join models
+        for (int i = 0, n = jm.size(); i < n; i++) {
+            // get primary model
+            QueryModel jm1 = jm.getQuiet(i);
+            jm1 = jm1 != null ? jm1.getNestedModel() : null;
+            if (jm1 == null) {
+                return;
+            }
+            // get secondary model
+            QueryModel jm2 = jm1.getJoinModels().getQuiet(1);
+            // if order by advice has no table prefixes, we preserve original behaviour and pass it on.
+            if (!orderByAdviceHasDot) {
+                setAndCopyAdvice(jm1, orderByAdvice, orderByMnemonic, orderByDirectionAdvice);
+                optimiseOrderBy(jm1, orderByMnemonic);
+                return;
+            }
+            // if the order by advice is for more than one table, don't propagate it, as a sort will be needed anyway
+            if (!checkForConsistentPrefix(orderByAdvice)) {
+                return;
+            }
+            // if the orderByAdvice prefixes do not match the primary table name, don't propagate it
+            final CharSequence prefix = getTablePrefix(orderByAdvice.getQuick(0).token);
+            if (!(Chars.equalsNc(prefix, jm1.getTableName())
+                    || (jm1.getAlias() != null && Chars.equalsNc(prefix, jm1.getAlias().token)))) {
+                optimiseOrderBy(jm1, orderByMnemonic);
+                return;
+            }
+            // order by advice is pushable, so now we copy it and strip the table prefix
+            if (advice == null) {
+                advice = duplicateAdviceAndTakeSuffix();
+            }
+            // if there's a join, we need to handle it differently.
+            if (jm2 != null) {
+                final int joinType = jm2.getJoinType();
+                switch (joinType) {
+                    case QueryModel.JOIN_ASOF:
+                        // For asof join, we only propagate advice if its ordered beginning with the designated timestamp
+                        CharSequence token = advice.getQuick(0).token;
+                        QueryColumn qc = jm1.getAliasToColumnMap().get(token);
+                        // if there is a matching column, and it is the designated timestamp, then propagate advice
+                        if (qc != null
+                                && qc.getColumnType() == ColumnType.TIMESTAMP
+                                && Chars.equalsIgnoreCase(jm1.getTimestamp().token, qc.getAst().token)) {
+                            setAndCopyAdvice(jm1, advice, orderByMnemonic, orderByDirectionAdvice);
+                        }
+                        break;
+                    case QueryModel.JOIN_INNER:
+                        // for inner, we propagate on both sides
+                        setAndCopyAdvice(jm1, advice, orderByMnemonic, orderByDirectionAdvice);
+                        setAndCopyAdvice(jm2, advice, orderByMnemonic, orderByDirectionAdvice);
```

Comment:
Why do we pass the order by to the second table? If I comment this line, all the SqlOptmiserTests still pass.